shell/common/pointer_data_dispatcher.h

```diff
@@ -0,0 +1,176 @@
+///       platform thread, and sends it to the UI thread for the final
+///       construction of the `PointerDataDispatcher`.
+class PointerDataDispatcher {
+ public:
+  /// The interface for Engine to implement.
+  class Delegate {
+   public:
+    /// Actually dispatch the packet using Engine's `animator_` and
+    /// `runtime_controller_`.
+    virtual void DoDispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                                  uint64_t trace_flow_id) = 0;
+
+    //----------------------------------------------------------------------------
+    /// @brief    Schedule a secondary callback to be executed right after the
+    ///           main `VsyncWaiter::AsyncWaitForVsync` callback (which is added
+    ///           by `Animator::RequestFrame`).
+    ///
+    ///           Like the callback in `AsyncWaitForVsync`, this callback is
+    ///           only scheduled to be called once, and it's supposed to be
+    ///           called in the UI thread. If there is no AsyncWaitForVsync
+    ///           callback (`Animator::RequestFrame` is not called), this
+    ///           secondary callback will still be executed at vsync.
+    ///
+    ///           This callback is used to provide the vsync signal needed by
+    ///           `SmoothPointerDataDispatcher`.
+    virtual void ScheduleSecondaryVsyncCallback(
+        std::function<void()> callback) = 0;
+  };
+
+  //----------------------------------------------------------------------------
+  /// @brief      Signal that `PlatformView` has a packet to be dispatched.
+  ///
+  /// @param[in]  packet             The `PointerDataPacket` to be dispatched.
+  /// @param[in]  trace_flow_id      The id for `Animator::EnqueueTraceFlowId`.
+  virtual void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                              uint64_t trace_flow_id) = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Default destructor.
+  virtual ~PointerDataDispatcher();
+};
+
+//------------------------------------------------------------------------------
+/// The default dispatcher that forwards the packet without any modification.
+///
+class DefaultPointerDataDispatcher : public PointerDataDispatcher {
+ public:
+  DefaultPointerDataDispatcher(Delegate& delegate) : delegate_(delegate) {}
+
+  // |PointerDataDispatcer|
+  void DispatchPacket(std::unique_ptr<PointerDataPacket> packet,
+                      uint64_t trace_flow_id) override;
+
+  virtual ~DefaultPointerDataDispatcher();
+
+ protected:
+  Delegate& delegate_;
+
+  FML_DISALLOW_COPY_AND_ASSIGN(DefaultPointerDataDispatcher);
+};
+
+//------------------------------------------------------------------------------
+/// The dispatcher that filters out irregular input events delivery to provide
+/// a smooth scroll on iPhone X/Xs.
+///
+/// This fixes https://github.com/flutter/flutter/issues/31086.
```

Comment:
I don't think this is necessary. You have provided all details necessary in the documentation (if you don't think you have, please elucidate it here). Git archaeologists will be able to get to the bug from the commit anyway.